### ISSUE TITLE
Checking muPC Input Mode

### ISSUE THREAD

**Issue by AndrewEdmonds11** [alcap-org/g4sim#47]
One of the tasks in issue #46 is to check the "muPC" input mode of g4sim.

The first thing I have realised is that I was calculating the fraction of stopped muons like so:

```
fraction stop = (n_stopped_muons_in_target) / (n_input_muons)
```

when the data calculates it with `n_muons_in_muSc` instead of `n_input_muons`. In a short 500k run these were 392000 and 500000 respectively (approx 20% difference) which does invalidate my validation.

Anyway, I've set some of the volumes in the beamline to be monitors and added a few more monitor planes.

Here is a plot of the z-position (y-axis) vs the volume name (x-axis):
![hzcoords](https://cloud.githubusercontent.com/assets/1848575/5183574/30c29560-74a8-11e4-9f69-f8da580f3a62.png)

Here the z-axis is the number of muons in that volume, so projecting down to the x-axis we can see how many muons are in each volume:
![hnumbers](https://cloud.githubusercontent.com/assets/1848575/5183585/5134491a-74a8-11e4-8458-7712897ea04e.png)

(MuSCA has a hole in it so that explains why it has so many fewer muons in it)

Something I'm unsure of is that it looks like the number of muons lost at MuSC is the same as in MuSCA (the numbers don't match up exactly). I've been thinking that the MuSC and MuSCA are the same size but MuSCA has a hole in (this is what's in the simulation). Looking at the verbose output of a very short run, it seems that muons go through either the MuSC or MuSCA.

Now, plotting the momentum against the volume name we have the following (NB log scale):
![hmomentum_logz](https://cloud.githubusercontent.com/assets/1848575/5183687/2af8130c-74a9-11e4-8a6a-9671a3212342.png)

Here we can see we got a small (~5 MeV/c) drop going through the MuSC and as we travel down the beam port we see a gradual spread downwards, presumably from muons scattering around the walls of the port. Does this make sense to others? Are the drops too large?

**Next Steps**
- check what's going on with MuSC and MuSCA
- check that the drops in momentum aren't too high

**Comment by AndrewEdmonds11** [comment]
Ok, I've made some progress with this and managed to tune this input mode to match the percentage of muons stopping in the Al50 target by changing the thickness of the MuFoil volume from 0.075 mm to 0.02 mm. 

This volume is placed in the MuTrigger volume (the volume that contains all the muSc/muPC etc.) 43.8875 mm upstream of the centre before the muSc and muPC.

I'm not sure if the MuFoil was ever measured or if the 0.075 was just a rough guess. Let me know if this is not the case.

For Al50, I got the percentage of stopped muons being (6.47 +- 0.04)% and in the data it's (6.4 +- 0.1)%
Unfortunately, the Al100 run I did gave me 13.7% stopped muons when the data says that we had 11.3% so there may be a few more tweaks to make.

Anyway, here's some more details of the runs I did with some comments from me. They were all 500k muon runs in Al50.

| commit | Initial Muon Position (relative to centre of MuTrigger) [mm] | No. of Stopped Muons | No. of Muons in MuSC | % of stopped muons | Comment |
|----------|---------------------------------------------------------------------|------------------------------|-----------------------------------|-----------------------------|-----------------|
59a14d7 | -31 | 17297 | 499992 | 3.46 | location of the MuSC |
34b0d67 | -32 | 18088 | 499993 | 3.62 | just before the muSc |
b542625 | -40 | 17382 | 404310 | 4.30 | |
be5c82f | -45 | 47656 | 403675 | 11.58 | |
37fbe1b | -43 | 18070 | 404015 | 4.47 | just after MuFoil |
a7c4a30 | -44 | 47730 | 403777 | 11.82 | just before MuFoil |

At this point, I thought that the thickness of the MuFoil meant that the muons were losing too much energy as they passed through it and so were more likely to stop in the target so I started changing its thickness (all positions are 44 mm behind the centre of the MuTrigger):

| commit | Thickness of MuFoil (made of mylar) [mm] | No. of Stopped Muons | No. of Muons in MuSC | % of stopped muons | Comment |
|----------|---------------------------------------------------------------------|------------------------------|-----------------------------------|-----------------------------|-----------------|
a7c4a30 | 0.075 | 47730 | 403777 | 11.82 | same as last entry in the table above |
8fe3956 | 0.05 | 39103 | 403946 | 9.68 | |
aa651ec | 0.025 | 282720 | 403999 | 7.11 | |
1c724f1 | 0.01 | 22247 | 404165 | 5.50 | |
51e24e2 | 0.02 | 26164 | 404210 | 6.47 | matches the data value of (6.4 +- 0.1)% |

**Comment by jrquirk** [comment]
The MuFoil is the Al wrapping on the upstream scints? If so I can ask a person who works on MuSun sims.

**Comment by AndrewEdmonds11** [comment]
I don't think so. It's just a 100x100x0.075mm^3 sheet of Mylar placed before the MuSC. Regardless, if you can get some more information from the MuSun people that would be very useful :)

**Comment by jrquirk** [comment]
So it's a fictitious degrader inserted there to help reproduce the data in the simulations as far as we know?

**Comment by AndrewEdmonds11** [comment]
Don't know. I assumed it exists in reality but that we may be a bit uncertain of the thickness.
